plugins/clipboard/plugin.js

```diff
@@ -2539,6 +2580,231 @@
+
+	CKEDITOR.plugins.clipboard.fallbackDataTransfer.prototype = {
+		/**
+		 * Whether {@link CKEDITOR.plugins.clipboard.fallbackDataTransfer fallbackDataTransfer object} should
+		 * be used when operating on native `dataTransfer`. If `true` is returned, it means custom MIME types
+		 * are not supported in the current browser (see {@link #_isCustomMimeTypeSupported}).
+		 *
+		 * @returns {Boolean}
+		 */
+		isRequired: function() {
+			var fallbackDataTransfer = CKEDITOR.plugins.clipboard.fallbackDataTransfer;
+
+			if ( fallbackDataTransfer._isCustomMimeTypeSupported === null ) {
+				// If there is no `dataTransfer` we cannot detect if fallback is needed.
+				// Method returns `false` so regular flow will be applied.
+				if ( !this._nativeDataTransfer ) {
+					return false;
+				} else {
+					var testValue = 'cke test value',
+						testType = 'cke/mimetypetest';
+
+					fallbackDataTransfer._isCustomMimeTypeSupported = false;
+
+					try {
+						this._nativeDataTransfer.setData( testType, testValue );
+						fallbackDataTransfer._isCustomMimeTypeSupported = this._nativeDataTransfer.getData( testType ) === testValue;
+						this._nativeDataTransfer.clearData( testType );
+					} catch ( e ) {}
+				}
+			}
+			return !fallbackDataTransfer._isCustomMimeTypeSupported;
+		},
+
+		/**
+		 * Returns the data of the given MIME type if stored in a regular way or in a special comment. If given type
+		 * is the same as {@link #_customDataFallbackType} the whole data without special comment is returned.
+		 *
+		 * @param {String} type
+		 * @returns {String}
+		 */
+		getData: function( type ) {
+			var value = null;
+
+			// If we are getting the same type which may store custom data we need to extract content only.
+			if ( type === this._customDataFallbackType ) {
+				value = this._extractDataComment( this._getData( type ) ).content;
+
+			// If we are getting different type we need to check inside data comment if it is stored there.
+			} else {
+				var result = this._extractDataComment( this._getData( this._customDataFallbackType ) );
+				if ( result.data && result.data[ type ] ) {
+					value = result.data[ type ];
+
+				// And then fallback to regular `getData`.
+				} else {
+					value = this._getData( type );
+				}
+			}
+
+			return value !== null ? value : '';
+		},
+
+		/**
+		 * Sets given data in native `dataTransfer` object. If given MIME type is not supported it uses
+		 * {@link CKEDITOR.plugins.clipboard.fallbackDataTransfer#_customDataFallbackType} MIME type
+		 * to save data using special comment format:
+		 *
+		 * 		<!--cke-data:{ type: value }-->
+		 *
+		 * It is important to keep in mind that `{ type: value }` object is stringified (using `JSON.stringify`)
+		 * and encoded (using `encodeURIComponent`).
+		 *
+		 * @param {String} type
+		 * @param {String} value
+		 */
+		setData: function( type, value ) {
+			// Extract custom data if used type is the same one which is used for storing it.
```

Comment:
I think this code is just a bit misleading: it's not only extracting, but also appending this data.

How about something like that?
> // In case of setting type that stores custom data, make sure that current custom data gets applied.